File: Java-8/algorithms/src/test/java/org/gb/sample/algo/PermutationsTest.java
package org.gb.sample.algo;

import java.util.Arrays;
import java.util.List;
import java.util.Set;

import org.apache.log4j.Logger;
import org.gb.sample.algo.permutation.Permutations;
import org.junit.Assert;
import org.junit.Test;

public class PermutationsTest {

	private static final Logger LOGGER = Logger.getLogger(PermutationsTest.class);

	@Test
	public void permute() {
		List<Character> characters = Arrays.asList('a', 'b', 'c', 'd');
		Set<String> permutations = Permutations.permute(characters);

		Assert.assertEquals(24, permutations.size());
	}

	@Test
	public void permuteAndReturnList() {
		List<Character> characters = Arrays.asList('a', 'b', 'c', 'd', 'e');
		List<String> permutations = Permutations.permuteAndReturnList(characters);

		Assert.assertEquals(120, permutations.size());
	}

	@Test
	public void permuteParallel() {
		List<Character> characters = Arrays.asList('a', 'b', 'c', 'd', 'e');
		Set<String> permutations = Permutations.permuteParallel(characters);

		Assert.assertEquals(120, permutations.size());
	}	
}

File: CoreScala/simple-scala/src/test/scala/org/gb/sample/scala/PermutationsTest.scala
package org.gb.sample.scala

import scala.collection.mutable.IndexedSeq
import org.scalatest.funsuite.AnyFunSuite

class PermutationsTest extends AnyFunSuite {
    test("permutations of 5 distinct items") {
        val characters = IndexedSeq('a', 'b', 'c', 'd', 'e')
        val permutations = Permutations.permute(characters);
        assert(permutations.size == 120)
    }
}